base/reflection.jl

```diff
@@ -754,6 +754,8 @@ function ismutationfree(@nospecialize(t))
         return datatype_ismutationfree(t)
     elseif isa(t, Union)
         return ismutationfree(t.a) && ismutationfree(t.b)
+    elseif isa(t, Core.TypeofVararg)
+        return true
```

Comment:
```suggestion
        isdefined(t, :T) && !ismutationfree(t.T) && return false
        return true
```
would be better?

Reply:
The added test failed on CI
. Just find that the branch should be `t === Core.TypeofVararg` instead, so the fix should be moved to c side?

Reply:
Do you mean the failure is because:
```julia
julia> CC.datatype_ismutationfree(Core.TypeofVararg)
false
```
? Then we may want to modify the flag in C side as we did in e.g. #48868.